eppoclient/httpclient.go

```diff
@@ -46,18 +49,23 @@ func (hc *httpClient) get(resource string) string {
 	req.URL.RawQuery = q.Encode()
 
 	resp, err := hc.client.Do(req)
-
 	if err != nil {
-		log.Fatal(err)
+		return "", err // Return an empty string and the error
 	}
+	defer resp.Body.Close() // Ensure the response body is closed
 
 	if resp.StatusCode == 401 {
 		hc.isUnauthorized = true
+		return "", fmt.Errorf("unauthorized access") // Return an error indicating unauthorized access
+	}
+
+	if resp.StatusCode >= 500 {
+		return "", fmt.Errorf("server error: %d", resp.StatusCode) // Handle server errors (status code > 500)
```

Comment:
Another approach we can take here is to return "positive" when http is 2xx and have a generic error message otherwise.